### PR TITLE
KO-285: Allow Operator operations with failed pods

### PR DESCRIPTION
**Pull request by abhishekdwivedi3060**

### CODE REVIEW

File: test/cluster_test.go [hunk 1]
```diff
@@ -45,6 +45,11 @@ var _ = Describe(
 		// 		DeployClusterWithSyslog(ctx)
 		// 	},
 		// )
+		Context(
+			"DeployClusterWithSyslog", func() {
```
Comment (abhishekdwivedi3060) [inline test/cluster_test.go:49]: Will fix this typo in next commit.

File: test/cluster_test.go [hunk 2]
```diff
@@ -124,6 +129,92 @@ func ScaleDownWithMigrateFillDelay(ctx goctx.Context) {
...
+					aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+					Expect(err).ToNot(HaveOccurred())
+					aeroCluster.Spec.Image = baseImage + ":6.4.0.4"
+					err = updateCluster(k8sClient, ctx, aeroCluster)
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Scale up")
+					aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+					Expect(err).ToNot(HaveOccurred())
+					aeroCluster.Spec.IgnorePodList = []string{ignorePodName}
```
Comment (tanmayja) [inline test/cluster_test.go:201]: any reason of setting it again after setting it at line 179? AFAIK only user can remove the name from IgnorePodList right?

File: test/cluster_test.go [hunk 3]
```diff
@@ -124,6 +129,92 @@ func ScaleDownWithMigrateFillDelay(ctx goctx.Context) {
 	)
 }
 
+func clusterWithIgnorePodList(ctx goctx.Context) {
```
Comment (tanmayja) [inline test/cluster_test.go:132]: can we add a test after removing the name from list and make that pod running to check if updated config is correctly reflected on that pod as well?

File: test/cluster_test.go (hunk 2 above)
Comment (abhishekdwivedi3060) [reply test/cluster_test.go:201]: Not required.

File: controllers/aero_info_calls.go [hunk 4]
```diff
@@ -18,6 +18,7 @@ import (
 	"time"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
```
[the remark this answers is not in the document]
Comment (abhishekdwivedi3060) [reply controllers/aero_info_calls.go:21]: golang-set is only used in SC file for management-lib changes. 
As discussed, we won't change existing code to golang-set in this PR, so adding new changes with golang-set was causing confusion. So kept everything on apimachinery set pkg and will move to golang-set in one go.

File: controllers/pod.go [hunk 5]
```diff
@@ -345,9 +350,14 @@ func (r *SingleClusterReconciler) ensurePodsRunningAndReady(podsToCheck []*corev
 	return reconcileRequeueAfter(10)
 }
 
-func getFailedAndActivePods(pods []*corev1.Pod) (failedPods, activePods []*corev1.Pod) {
+func getFailedAndActivePods(pods []*corev1.Pod, ignorablePodNames sets.Set[string],
+) (failedPods, activePods []*corev1.Pod) {
 	for idx := range pods {
 		pod := pods[idx]
+		if ignorablePodNames.Has(pod.Name) {
```
Comment (tanmayja) [inline controllers/pod.go:357]: we calls `getFailedAndActivePods` at 4 places, at 2 places we make sure that there are no ignorable pods present in pods list passed to this function, and in other two places from `reconcileRacks()`, there are chances of having ignorable pods in pods list. Can we write it better and remove check `if ignorablePodNames.Has(pod.Name) {` from `getFailedAndActivePods` if possible.

**Pull request merged**